plugins/teststeps/waitport/waitport.go

```diff
@@ -0,0 +1,244 @@
+// Package waitport provides an ability of waiting for a port on remote host to be opened for listening
+package waitport
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/facebookincubator/contest/pkg/event"
+	"github.com/facebookincubator/contest/pkg/event/testevent"
+	"github.com/facebookincubator/contest/pkg/target"
+	"github.com/facebookincubator/contest/pkg/test"
+	"github.com/facebookincubator/contest/pkg/xcontext"
+	"github.com/facebookincubator/contest/plugins/teststeps"
+)
+
+// Name is the name used to look this plugin up.
+const Name = "waitport"
+
+// event names for this plugin.
+const (
+	EventCmdStart = event.Name(Name + "Start")
+	EventCmdEnd   = event.Name(Name + "End")
+)
+
+// Events defines the events that a TestStep is allow to emit
+var Events = []event.Name{
+	EventCmdStart,
+	EventCmdEnd,
+}
+
+// WaitPort provides an ability of waiting for a port on remote host to be opened for listening
+type WaitPort struct {
+}
+
+// Name returns the plugin name.
+func (ts *WaitPort) Name() string {
+	return Name
+}
+
+// Run executes the cmd step.
+func (ts *WaitPort) Run(ctx xcontext.Context, ch test.TestStepChannels, inputParams test.TestStepParameters, ev testevent.Emitter, resumeState json.RawMessage) (json.RawMessage, error) {
+	params, err := parseParameters(inputParams)
+	if err != nil {
+		return nil, err
+	}
+
+	f := func(ctx xcontext.Context, targetWithData *teststeps.TargetWithData) error {
+		target := targetWithData.Target
+		targetParams, err := expandParameters(target, params)
+		if err != nil {
+			return err
+		}
+
+		// Emit EventCmdStart
+		payload, err := json.Marshal(targetParams)
```

Comment:
Add a comment here that this can emit duplicate events on server restart / job resumption